contracts/LiquidityBridgeContract.sol

```diff
@@ -367,6 +368,32 @@ contract LiquidityBridgeContract {
         return transferredAmountOrErrorCode;
     }
 
+    function registerPegOut(
+        Quote memory quote,
+        bytes memory signature,
+        uint256 height
+    ) public noReentrancy returns (int256) {
+
+        // check quote.lbcAddress if its valid
+        //todo: should it be validateAndHashQuoteis enough or require(address(this) == quote.lbcAddress, "Wrong LBC address"); ?
+        bytes32 quoteHash = validateAndHashQuote(quote);
+
+        // check if signature is valid
+        require(SignatureValidator.verify(quote.liquidityProviderRskAddress, quoteHash, signature), "Invalid signature");
+
+        // quote.depositHeightLimit is less or equal to the current height
+        require(height <= block.number, "Block height overflown");
```

Comment:
You should check the timestamp and not the block number (check the flyover peg-out design and the comments from Vadym)

Reply:
replaced for `require(quote.timeForDeposit <= block.timestamp, "LBC: Block height overflown");`